Added tests for FAQ and positive tests for order registration

ncherny opened the pull request.

ncherny committed:
Added tests for FAQ and positive tests for order registration

okryukova commented on pom.xml:
⚠️Можно улучшить. Хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройку (если необходима) и как запускать (mvn clean test - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблицы используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

okryukova commented on src/main/java/ru/yandex/praktikum/pageObjectModels/MainPage.java:
```diff
@@ -0,0 +1,72 @@
+package ru.yandex.praktikum.pageObjectModels;
```
⛔️Нужно исправить. Необходимо исправить наименование пакетов в соответствии с правилами нейминга. Пакеты — допустимы только строчные буквы. В случае, если слов несколько, их следует разделять точкой. Префикс должен быть одним из доменных имен верхнего уровня, таких как com, ru и т.д. Наименование пакета должно отражать суть содержимого. https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

okryukova commented on src/test/java/MainPageFaqTests.java:
```diff
@@ -0,0 +1,62 @@
+        options.addArguments("--no-sandbox", "--headless", "--disable-dev-shm-usage");
+        driver = new ChromeDriver(options);
+
+        driver.get("https://qa-scooter.praktikum-services.ru/");
```
⚠️Можно улучшить. Url "https://qa-scooter.praktikum-services.ru/" обычно выделяют как константу (можно реализовать в классе MainPage).

okryukova commented on src/test/java/MainPageFaqTests.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Parameterized.Parameters
+    public static Object[][] getFaqData(){
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

okryukova commented on src/test/java/MainPageFaqTests.java:
```diff
@@ -0,0 +1,62 @@
+    @Parameterized.Parameters
+    public static Object[][] getFaqData(){
+        return new Object[][]{
+                {"Сколько это стоит? И как оплатить?", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
⚠️Можно улучшить. Тестовые данные  должны содержаться либо в константах тестового класса, либо в отдельном файле

okryukova commented on src/test/java/MainPageFaqTests.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @After
+    public void teardown() {
```
⚠️Можно улучшить. Блоки закрытия и открытия  браузера одинаковы для всех тестовых классов. Целесообразно вынести их в базовый или утилитный класс.

okryukova commented on src/test/java/OrderRegistrationTests.java:
```diff
@@ -0,0 +1,122 @@
+    private void registerOrderOnRegistrationPage() {
+        orderRegistrationPage.waitFirstPageToLoad();
+        orderRegistrationPage.setName(name);
+        orderRegistrationPage.setSurname(surname);
```
⚠️Можно улучшить. В теме UI-тестирование > Page Object > Работа с методами > Объединение методов хорошо описан прием объединения методов. Применяй его для сокращения кода автотеста, где это необходимо. Например, при заполнении форм.

okryukova commented on src/test/java/OrderRegistrationTests.java:
```diff
@@ -0,0 +1,122 @@
+
+        orderRegistrationPage.clickPlaceOrderButton();
+        orderRegistrationPage.waitOrderConfirmationPopUpToLoad();
+        orderRegistrationPage.clickConfirmOrderButton();
```
⛔️Нужно исправить. В каждом тесте должен быть ассерт . Добавь, пожалуйста

okryukova commented on src/test/java/OrderRegistrationTests.java:
```diff
@@ -0,0 +1,122 @@
+        orderRegistrationPage.clickConfirmOrderButton();
+    }
+
+    private void sendOrder() {
```
⛔️Нужно исправить. Неиспользуемый код нужно удалять

ncherny committed:
Fixed issues from comments

- Added more detailed readme file
- Changed package name to use '.' instead of camel case
- Moved page url to a constant
- Added 'name' argument with test data to tests
- Migrated project to junit 5 from junit 4
- Moved test data to csv files
- Created a base test class for before and after blocks
- Moved registerOrderOnRegistrationPage method from test class to
  OrderRegistrationPage page object model class
- Added asserts to registration tests
- Removed redundant code
